framework/build.mk

```diff
@@ -46,9 +46,15 @@ ifneq (x$(MOOSE_NO_PERF_GRAPH), x)
 endif
 
 ifneq ($(GPERF_DIR), )
+ifneq (,$(findstring prof,$(METHOD)))
     libmesh_CXXFLAGS += -DHAVE_GPERFTOOLS -I$(GPERF_DIR)/include
     libmesh_LDFLAGS := -L$(GPERF_DIR)/lib -ltcmalloc_and_profiler $(libmesh_LDFLAGS)
 endif
+ifeq (opt,$(METHOD))
```

Comment:
The following two lines are the same as the above two lines. 

Is it possible to combine them? Introduce an `or` operator?

Reply:
I am no `Makefile` expert. This was the best stackoverflow post I found but it doesn't look great to me and it's more simple than our case: https://stackoverflow.com/questions/7656425/makefile-ifeq-logical-or

Reply:
Better to issue an error in other modes to let users know they should not expect gperftools to work in those modes.

Reply:
Yeah - we should document this behavior in the profiling docs - that perf tools/tcmalloc are not linked when building in dbg/devel mode regardless of whether gperf_dir is set.

Reply:
Ok I removed duplication, added an error, and updated profiling documentation